### ISSUE TITLE
Cannot debug mips binaries with current master

### ISSUE THREAD

**Issue by arichardson** [CTSRD-CHERI/gdb#27]
With the current master branch I get the following when attempting to debug a dynamically linked MIPS binary. It works with da9e0215d1831b9b87287f216c7b5fd72a1b3acc

```
root@qemu-cheri128-alex:~ # env LIBUNWIND_PRINT_APIS=1 gdb --args /buildroot/libunwind-mips-build/lib/test-simple_exceptions2-static-O3
GNU gdb (GDB) 8.3
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "mips64-unknown-freebsd13".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
/Users/alex/cheri/gdb/gdb/mips-tdep.c:9590: internal-error: struct gdbarch *mips_gdbarch_init(struct gdbarch_info, struct gdbarch_list *): Assertion `mips_regnum.cap0 != -1' failed.
A problem internal to GDB has been detected,
further debugging may prove unreliable.
Quit this debugging session? (y or n)
```

**Comment by bsdjhb** [comment]
Aww geeze.  I missed an obvious 'break'.

**Comment by bsdjhb** [comment]
Fixed in a6937c9
